### PR TITLE
Add On Disk Database Support

### LINKED ISSUE #332: On Disk Database.
Enable Convoy run without MongoDB. Use an on-disk database. -- BoltDB.

### PR DESCRIPTION
**Pull request by jirevwe**
Resolves #332

### CODE REVIEW

File: datastore/mongo/application.go [hunk 1]
```diff
@@ -64,15 +60,15 @@ func (db *appRepo) LoadApplicationsPaged(ctx context.Context, groupID string, pa
...
 		if err != nil {
 			log.Errorf("failed to count events in %s. Reason: %s", app.UID, err)
-			return apps, pager.PaginationData{}, err
+			return apps, models.PaginationData{}, err
 		}
 		apps[i].Events = count
 	}
 
-	return apps, paginatedData.Pagination, nil
+	return apps, models.PaginationData(models.PaginationData(paginatedData.Pagination)), nil
```
[the remark this answers is not in the document]
Comment (jirevwe) [reply datastore/mongo/application.go:68]: Ops, that's a mistake

[the remark this answers is not in the document]
Comment (danvixent) [reply datastore/mongo/application.go:68]: nice, tha pr looks sweet.

File: cmd/main.go [hunk 2]
```diff
@@ -83,7 +82,7 @@ func main() {
 				return err
 			}
 
-			db, err = datastore.New(cfg)
+			db, err = mongo.New(cfg)
```
Comment (ogbanugot) [inline cmd/main.go:85]: shouldn't the choice of mongo here be based on the specs in cfg?

Comment (jirevwe) [reply cmd/main.go:85]: Yes, but I'll add the switch later Everything would be ready by tomorrow. I don't want to break implementation which is why I defaulted to mongo

Comment (ogbanugot) [reply cmd/main.go:85]: ohh okay, i get. Nice one.

**Review by Subomi — APPROVED** [review]

**Pull request merged**